### PR TITLE
 WIP initial phrase suggester refactoring

### PR DESCRIPTION
**Pull request by cbuescher**
This is a very early WIP that add basic serialization and tests to the PhraseSuggestionBuilder, inteded to initial discussion.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/search/suggest/SuggestBuilder.java [hunk 1]
```diff
@@ -119,14 +129,35 @@ public T text(String text) {
...
+
         protected void setRegex(String regex) {
             this.regex = regex;
         }
 
+        /**
+         * get the regex for this suggestion
+         */
+        public String regEx() {
+            return this.prefix;
```
Comment (colings86) [inline core/src/main/java/org/elasticsearch/search/suggest/SuggestBuilder.java]: This should probably return regex not prefix?

File: core/src/test/java/org/elasticsearch/search/suggest/SuggestionBuilderTestCase.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import org.elasticsearch.test.ESTestCase;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.not;
+
+public abstract class SuggestionBuilderTestCase<SB extends SuggestionBuilder<SB>> extends ESTestCase {
```
Comment (colings86) [inline core/src/test/java/org/elasticsearch/search/suggest/SuggestionBuilderTestCase.java]: Can we call this `AbstractSuggestionBuilderTestCase` to match with `AbstractQueryTestCase`?

**Comment by colings86** [comment]
Left two minor comments but LGTM

**Comment by cbuescher** [comment]
Closing and will reopen again vs. new feature branch.

**Pull request closed without merging**